.github/workflows/typedoc-generate-gitbook-docs.yml

```diff
@@ -0,0 +1,54 @@
+name: Deploy TypeDoc to GitBook
+
+on:
+  push:
+    branches:
+      - develop
+  
+env:
+  NODE_VERSION: 18.x
+  ENTRY_FILE: "packages"
+  CONFIG_PATH: "tsconfig.base.json"
+  USES_PNPM: "true"
+  
+jobs:
+  deploy:
+    concurrency: ci-${{ github.ref }}
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+  
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+  
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9
+  
+      - name: Install dependencies
+        run: pnpm install --no-frozen-lockfile
+    
+      - name: Build project
+        run: pnpm build
+    
+      - name: Build documentation
+        run: pnpm docs:generate-markdown
+    
+      - name: Deploy to GitBook
+        env:
+          GITBOOK_TOKEN: ${{ secrets.GITBOOK_TOKEN }}
+          GH_PAT: ${{ secrets.GH_PAT }}
```

Comment:
There seems to be no point in loading the secret into the env as the key is directly used in the `git clone` / `git push` commands. Right?